### PR TITLE
Adding `python setup.py test --figure-only`

### LINKED ISSUE #2556: Adding `python setup.py test --figure-only`
We have several flags that can be passed into `python setup.py test` to trigger tests.
There are several tests that create figures and we would like to be able to run just those instead of the entire test suite. 

This is a guess:
It would require the combination of adding support for the flag (`--figure-only`) within this [file.](https://github.com/sunpy/sunpy/blob/master/sunpy/tests/__init__.py)

Also figuring out the correct combination to pass to pytest in order to just run the figure tests. This could be made setting flags [here](https://github.com/sunpy/sunpy/blob/master/sunpy/tests/setup_command.py) or something much more complicated.

### PR DESCRIPTION
**Pull request by himanshukgp**
Closes #2556

### CODE REVIEW

**Comment by himanshukgp** [comment]
This works for `sunpy.self_test`

File: sunpy/tests/__init__.py [hunk 1]
```diff
@@ -81,8 +81,13 @@ def main(modulename='', coverage=False, cov_report=False,
         all_args.append('--remote-data')
     if not offline:
         all_args.append('-k remote_data')
+
+    if figure_only:
+        figure = True
     if not figure:
         all_args.append('-m not figure')
+    if figure_only:
+        all_args.append('-m figure')
```
Comment (nabobalis) [inline sunpy/tests/__init__.py]: Why is the if statement repeated twice?

**Comment by nabobalis** [comment]
So this only adds it to sunpy.self_test()?

File: sunpy/tests/__init__.py (hunk 1 above)
Comment (himanshukgp) [reply sunpy/tests/__init__.py]: I added one more argument to only select figure.I left the previous case to include figure untouched.

**Comment by himanshukgp** [comment]
I am still working on the `setup.py` part.

File: sunpy/tests/__init__.py [hunk 2]
```diff
@@ -17,8 +17,8 @@
 
 
 def main(modulename='', coverage=False, cov_report=False,
-         online=False, offline=True, remote_data=False, figure=False, verbose=False,
-         parallel=0, args=None):
+         online=False, offline=True, remote_data=False, figure=False, figure_only=False,
+         verbose=False, parallel=0, args=None):
```
Comment (nabobalis) [inline sunpy/tests/__init__.py]: Would need the new keyword being added to the documentation string below.

Comment (himanshukgp) [reply sunpy/tests/__init__.py]: Yeah I'll add it in my next commit. Thanks.

File: sunpy/tests/__init__.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/tests/__init__.py]: Could the two `if figure_only:` statements not be merged?

**Comment by Cadair** [comment]
Can you also modify the .travis.yml file to use this flag for the figure tests please?

**Comment by himanshukgp** [comment]
Should I add this code [here](https://github.com/sunpy/sunpy/blob/master/.travis.yml#L86) ?
```
        python: 2.7
           stage: Comprehensive tests
           env: SETUP_CMD='test --figure-only --coverage' CONDA_DEPENDENCIES=''```

**Comment by nabobalis** [comment]
Looks right. Also for the py 3.5 one.

File: .travis.yml [hunk 3]
```diff
@@ -84,11 +84,19 @@ matrix:
...
+
          # Python 3.5 figure tests
          - python: 3.5
            stage: Comprehensive tests
            env: SETUP_CMD='test --figure --coverage' CONDA_DEPENDENCIES=''
 
+         - python: 3.5
+           stage: Comprehensive tests
+           env: SETUP_CMD='test --figure-only --coverage' CONDA_DEPENDENCIES=''
+
```
Comment (nabobalis) [inline .travis.yml]: We already have figure tests, we probably want to make those figure only

**Comment by himanshukgp** [comment]
@nabobalis Is it ok now ?

**Comment by nabobalis** [comment]
Yeah that looks good.

File: sunpy/tests/setup_command.py [hunk 4]
```diff
@@ -117,6 +122,7 @@ def generate_testing_command(self):
                        online=online,
                        offline=offline,
                        figure=self.figure,
```
Comment (nabobalis) [inline sunpy/tests/setup_command.py]: I wonder if all of these should be self.*

Comment (himanshukgp) [reply sunpy/tests/setup_command.py]: In the line 104 above, we did this
`online = self.online`

Comment (nabobalis) [reply sunpy/tests/setup_command.py]: So we could do the same for figure?

Comment (himanshukgp) [reply sunpy/tests/setup_command.py]: I don't see any problem if we do same for figure and figure_only. Should I ?

Comment (nabobalis) [reply sunpy/tests/setup_command.py]: Sure.

**Comment by Cadair** [comment]
I gave the online build a kick to see if we can get it to pass.

**Comment by nabobalis** [comment]
I think we should add a changelog entry for this.

**Comment by nabobalis** [comment]
CI is the known two test failures fixed in another PR.

**Pull request merged**